x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/EnterpriseSearch.java

```diff
@@ -40,4 +63,43 @@ protected XPackLicenseState getLicenseState() {
         // Register new actions here
         return Collections.emptyList();
     }
+
+    @Override
+    public Collection<Object> createComponents(
+        Client client,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ResourceWatcherService resourceWatcherService,
+        ScriptService scriptService,
+        NamedXContentRegistry xContentRegistry,
+        Environment environment,
+        NodeEnvironment nodeEnvironment,
+        NamedWriteableRegistry namedWriteableRegistry,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<RepositoriesService> repositoriesServiceSupplier,
+        Tracer tracer,
+        AllocationService allocationService
+    ) {
+        if (enabled == false) {
+            return Collections.emptyList();
+        }
+        final EngineIndexService engineService = new EngineIndexService(client, null);
+        return Collections.singletonList(engineService);
+    }
+
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+        return Arrays.asList(EngineIndexService.getSystemIndexDescriptor());
+    }
+
+    @Override
+    public String getFeatureName() {
+        return "ent-search";
```

Comment:
Is it best practice to have the feature name hard coded in this method, or should we create a `private final String featureName` that then gets returned with this method?